File: adapter/src/main/java/io/prizy/adapters/contest/mapper/MerchantMapper.java
package io.prizy.adapters.contest.mapper;

import io.prizy.adapters.contest.persistence.entity.MerchantEntity;
import io.prizy.domain.contest.model.Merchant;
import lombok.experimental.UtilityClass;

/**
 * @author Nidhal Dogga
 * @created 4/24/2022 1:28 PM
 */

@UtilityClass
public class MerchantMapper {

  public Merchant map(MerchantEntity entity) {
    return new Merchant(
      entity.getName(),
      entity.getSiren(),
      entity.getCapital(),
      entity.getAddress(),
      entity.getLogoMediaId()
    );
  }

  public MerchantEntity map(Merchant merchant) {
    return MerchantEntity.builder()
      .name(merchant.name())
      .siren(merchant.siren())
      .capital(merchant.capital())
      .address(merchant.address())
      .logoMediaId(merchant.logoMediaId())
      .build();
  }

}

File: jpa-adapter/src/main/kotlin/io/prizy/adapters/contest/mapper/MerchantMapper.kt
package io.prizy.adapters.contest.mapper

import io.prizy.adapters.contest.persistence.entity.MerchantEntity
import io.prizy.domain.contest.model.Merchant

/**
 *  @author Nidhal Dogga
 *  @created 4/24/2022 1:28 PM
 */

object MerchantMapper {

  fun map(entity: MerchantEntity): Merchant {
    return Merchant(
      name = entity.name,
      address = entity.address,
      siren = entity.siren,
      logoMediaId = entity.logoMediaId,
      capital = entity.capital,
    )
  }

  fun map(merchant: Merchant): MerchantEntity {
    return MerchantEntity(
      name = merchant.name,
      address = merchant.address,
      siren = merchant.siren,
      logoMediaId = merchant.logoMediaId,
      capital = merchant.capital,
    )
  }
}
